api/v1beta1/ibmvpcmachine_types.go

```diff
@@ -58,7 +58,28 @@ type IBMVPCMachineSpec struct {
 	PrimaryNetworkInterface NetworkInterface `json:"primaryNetworkInterface,omitempty"`
 
 	// SSHKeys is the SSH pub keys that will be used to access VM.
+	// Deprecated: Use SSHKeysRef instead
+	// +optional
 	SSHKeys []*string `json:"sshKeys,omitempty"`
+
+	// SSHKeysRef is the SSH pub keys that will be used to access VM.
+	// +optional
+	SSHKeysRef []*IBMVPCResourceReference `json:"sshKeysRef,omitempty"`
```

Comment:
One question here, why are both `SSHKeys` and `SSHKeysRef` marked as optional here? In a case where both are nil, it is currently not handled while creating the instance in `machine.go`-L150-174 below.

Reply:
got the reason, this comment can be ignored.